modules/gapi/src/backends/ie/giebackend.hpp

```diff
@@ -37,6 +38,37 @@ struct IECompiled {
     InferenceEngine::InferRequest      this_request;
 };
 
+// FIXME: Structure which collect all necessary sync primitives
+// will be deleted when the async request pool appears
+class SyncPrim {
+public:
+    void wait() {
+        std::unique_lock<std::mutex> l(m_mutex);
+        m_cv.wait(l, [this]{ return !m_is_busy; });
+    }
+
+    void release_and_notify() {
+        {
+            std::lock_guard<std::mutex> lock(m_mutex);
+            m_is_busy = false;
+        }
+        m_cv.notify_one();
+    }
+
+    void acquire() {
+        std::lock_guard<std::mutex> lock(m_mutex);
+        m_is_busy = true;
+    }
+
+private:
+    // To wait until the async request isn't over
+    std::condition_variable m_cv;
+    // To avoid spurious cond var wake up
+    bool m_is_busy = false;
+    // To sleep until condition variable wakes up
+    std::mutex m_mutex;
+};
```

Comment:
Please host it separately like we've done with `concurrent_shared_queue`.

Reply:
Do you mean to remove this `struct` ?

Reply:
Fixed